qiskit/circuit/random/utils.py

```diff
@@ -11,15 +11,394 @@
+    the interaction graph passed as argument.
+
+    This function will generate a random circuit by randomly selecting gates
+    from the set of standard gates in :mod:`qiskit.circuit.library.standard_gates`.
+    User can attach a float value indicating the probability of getting selected as a metadata to
+    the edge of the graph generated. If all the probabilities is passed as `None`, then the
+    probability of each qubit-pair of getting selected is set to 1/N.
+    (where N is the number of edges in the interaction_graph passed in)
+
+    If float values are present as probabilities but some are None, this will raise a ValueError.
+
+    If :arg:`max_operands` is set to 2, then in every iteration `N` 2Q gates and qubit-pairs
+    are chosen at random, the qubit-pairs are also chosen at random based on the probability
+    attached to the qubit-pair, the 2Q gates are applied on the qubit-paris which are unused
+    for that iteration, in case for a qubit-pair where target or control is already used up
+    for that iteration, if `insert_1q_oper` is enabled, then 1Q gates are applied to control or target
+    respectively, which are also chosen at random.
+
+    Example:
+
+    .. plot::
+       :include-source:
+
+       from qiskit.circuit.random.utils import random_circuit_from_graph
+       import rustworkx as rx
+       pydi_graph = rx.PyDiGraph()
+       pydi_graph.add_nodes_from(range(8))
+       cp_map = [(0, 2, 0.18), (1, 3, 0.15), (2, 4, 0.15), (3, 4, 0.22), (5, 7, 0.13), (4, 7, 0.17)]
+       pydi_graph.add_edges_from(cp_map)
+       inter_graph = (pydi_graph, None, None, None)
+       qc = random_circuit_from_graph(inter_graph, min_2q_gate_per_edge=1, measure = True)
+       qc.draw(output='mpl')
+
+    Args:
+        interaction_graph (int): Interaction Graph
+        min_2q_gate_per_edge (int): Minimum number of times every qubit-pair must be used.
+        max_operands (int): maximum qubit operands of each gate (between 1 and 2)
+        measure (bool): if True, measure all qubits at the end
+        conditional (bool): if True, insert middle measurements and conditionals
+        reset (bool): if True, insert middle resets. ( insert_1q_oper should be true )
+        seed (int): sets random seed (optional)
+        insert_1q_oper (bool): Insert 1Q operations to the qubits which are left after
+        applying a 2Q gate on the selected qubit-pair.
+        prob_conditional (float): Probability less than 1.0, this is used to control the
+        occurence of conditionals in the circuit.
+
+    Returns:
+        QuantumCircuit: constructed circuit
+
+    Raises:
+        CircuitError: when invalid options given
+        ValueError: when the probabilities for qubit-pairs are passed and they do not add to 1.0.
+        ValueError: when some edges have probability as None.
+    """
+    # Leaving out 'free_nodes'
+    pydi_graph, pydi_graph_node_map, _, _ = interaction_graph
+    edges_probs = pydi_graph.edges()
```

Comment:
`edges_probs` may also be an empty list, your code should handle that.

Reply:
done :)

Reply:
L204, it just returns circuit, made by `random_circuit`